src/compiler/preprocess/index.ts

```diff
@@ -83,6 +83,88 @@ async function replace_async(
 	return out.concat(final_content);
 }
 
+
+
+/**
+ * import decoded sourcemap from mozilla/source-map/SourceMapGenerator
+ * forked from source-map/lib/source-map-generator.js
+ * from methods _serializeMappings and toJSON
+ * we cannot use source-map.d.ts types, cos we access hidden properties
+ */
+function decoded_sourcemap_from_generator(generator: any) {
+	function areEqualMappings(a, b) {
+		return (
+			// sorted by selectivity
+			a.generatedColumn == b.generatedColumn &&
+			a.originalColumn == b.originalColumn &&
+			a.name == b.name &&
+			a.generatedLine == b.generatedLine &&
+			a.originalLine == b.originalLine &&
+			a.source == b.source
+		);
+	}
+	function convertMappings() {
+		// use global variable: generator
+		let previousGeneratedLine = 1;
```

Comment:
I think Conduitry was also asking if we could rename all the other variables and functions to use snake_case rather than camelCase

Reply:
good fight, good night
https://images-cdn.9gag.com/photo/a4ZXnVZ_700b.jpg

Reply:
I'm not quite sure how to interpret. I guess you are going to bed. Will you rename the variables later?

Reply:
I've pushed a commit to rename these and to also do some other tidying up and inlining of functions that are just called once. I'm aiming to merge this today and take a look for anything else screaming to be merged and then cut a release. Although this is technically a new feature, I'm thinking of still releasing this as a patch release.